pySDC/tests/test_sweepers/test_MPI_sweeper.py

```diff
@@ -0,0 +1,108 @@
+        else:
+            from pySDC.implementations.sweeper_classes.imex_1st_order import imex_1st_order as sweeper_class
+
+        from pySDC.implementations.problem_classes.HeatEquation_ND_FD import heatNd_forced as problem_class
+
+    dt = 1e-1
+    sweeper_params = {'num_nodes': num_nodes, 'quad_type': quad_type, 'QI': 'IEpar', 'QE': 'PIC'}
+    description = {}
+    description['problem_class'] = problem_class
+    description['sweeper_class'] = sweeper_class
+    description['sweeper_params'] = sweeper_params
+    description['level_params'] = {'dt': dt, 'residual_type': residual_type}
+    description['step_params'] = {'maxiter': 1}
+
+    controller = controller_nonMPI(1, {'logger_level': 30}, description)
+
+    if imex:
+        u0 = controller.MS[0].levels[0].prob.u_exact(0)
+    else:
+        u0 = np.ones_like(controller.MS[0].levels[0].prob.u_exact(0))
+    controller.run(u0, 0, dt)
+    controller.MS[0].levels[0].sweep.compute_end_point()
+    return controller.MS[0].levels[0]
+
+
+@pytest.mark.mpi4py
+@pytest.mark.parametrize("num_nodes", [2])
+@pytest.mark.parametrize("quad_type", ['GAUSS', 'RADAU-RIGHT'])
+@pytest.mark.parametrize("residual_type", ['last_abs', 'full_rel'])
+@pytest.mark.parametrize("imex", [True, False])
+def test_sweeper(num_nodes, quad_type, residual_type, imex, launch=True):
+    """
+    Make a test if the result matches between the MPI and non-MPI versions of a sweeper.
+    Tests solution at the right end point and the residual.
+
+    Args:
+        num_nodes (int): The number of nodes to use
+        quad_type (str): Type of nodes
+        residual_type (str): Type of residual computation
+        imex (bool): Use IMEX sweeper or not
+        launch (bool): If yes, it will launch `mpirun` with the required number of processes
+    """
+    if launch:
+        import os
+        import subprocess
+
+        # Set python path once
+        my_env = os.environ.copy()
+        my_env['PYTHONPATH'] = '../../..:.'
+        my_env['COVERAGE_PROCESS_START'] = 'pyproject.toml'
+
+        cmd = f"mpirun -np {num_nodes} python {__file__} --test_sweeper {num_nodes} {quad_type} {residual_type} {imex}".split()
+
+        p = subprocess.Popen(cmd, env=my_env, cwd=".")
+
+        p.wait()
+        assert p.returncode == 0, 'ERROR: did not get return code 0, got %s with %2i processes' % (
+            p.returncode,
+            num_nodes,
+        )
+    else:
+        import numpy as np
+
+        imex = False if imex == 'False' else True
+        MPI = run(use_MPI=True, num_nodes=int(num_nodes), quad_type=quad_type, residual_type=residual_type, imex=imex)
+        nonMPI = run(
+            use_MPI=False, num_nodes=int(num_nodes), quad_type=quad_type, residual_type=residual_type, imex=imex
+        )
+
+        assert np.allclose(MPI.uend, nonMPI.uend), 'Got different solutions at end point!'
+        assert np.allclose(MPI.status.residual, nonMPI.status.residual), 'Got different residuals!'
```

Comment:
Could you specify tight tolerances here, please? The default parameters are rather high, it seems.